src/commonTest/kotlin/fr/acinq/eclair/channel/states/ClosingTestsCommon.kt

```diff
@@ -1413,6 +1414,65 @@ class ClosingTestsCommon : EclairTestSuite() {
         }
     }
 
+    @Test
+    fun `recv BITCOIN_OUTPUT_SPENT (one revoked tx, counterparty published aggregated htlc tx)`() {
+        val (alice0, _, bobCommitTxs, htlcsAlice, htlcsBob) = prepareRevokedClose()
+
+        // bob publishes one of his revoked txs
+        val bobRevokedTx = bobCommitTxs[2]
+        assertEquals(8, bobRevokedTx.commitTx.tx.txOut.size)
+
+        val (alice1, _) = alice0.processEx(ChannelEvent.WatchReceived(WatchEventSpent(alice0.channelId, BITCOIN_FUNDING_SPENT, bobRevokedTx.commitTx.tx)))
+        assertTrue(alice1 is Closing)
+
+        // alice fetches information about the revoked htlcs
+        val htlcInfos = listOf(
+            ChannelAction.Storage.HtlcInfo(alice0.channelId, 4, htlcsAlice[0].paymentHash, htlcsAlice[0].cltvExpiry),
+            ChannelAction.Storage.HtlcInfo(alice0.channelId, 4, htlcsAlice[1].paymentHash, htlcsAlice[1].cltvExpiry),
+            ChannelAction.Storage.HtlcInfo(alice0.channelId, 4, htlcsBob[0].paymentHash, htlcsBob[0].cltvExpiry),
+            ChannelAction.Storage.HtlcInfo(alice0.channelId, 4, htlcsBob[1].paymentHash, htlcsBob[1].cltvExpiry),
+        )
+        val (alice2, _) = alice1.processEx(ChannelEvent.GetHtlcInfosResponse(bobRevokedTx.commitTx.tx.txid, htlcInfos))
+        assertTrue(alice2 is Closing)
+
+        // bob claims multiple htlc-timeout in a single transaction (this is possible with anchor outputs because signatures
+        // use sighash_single | sighash_anyonecanpay)
+        val bobHtlcTimeoutTxs = bobRevokedTx.htlcTxsAndSigs.filter { it.txinfo is Transactions.TransactionWithInputInfo.HtlcTx.HtlcTimeoutTx }
+        assertEquals(2, bobHtlcTimeoutTxs.size)
+        val bobHtlcTx = Transaction(
```

Comment:
Here Bob could also publish a tx that spends both his htlc-success txs ? Shouldn't we test for that too ?

Reply:
Yes he could, I can add a test but I feel it's redundant, we've tested elsewhere that we're able to claim revoked htlc-success, and we have a test that verifies that we scan all outputs (here and in `TransactionTestsCommon`).
Testing the htlc-success case here is a bit painful because we need to fetch the preimage to create valid htlc-success witnesses, it can be done but since it's already tested elsewhere I'm not sure it's worth it.

Reply:
I'll add one, it doesn't hurt to have more tests and not worth being lazy :)

Reply:
I saw the new transaction test  in TransactionTestsCommon where all htlc txs are aggregated but missed the one we have here ?

Reply:
Done in https://github.com/ACINQ/eclair-kmp/pull/236/commits/915c275886b640c645a51952ae5c28959480f315
It's even simpler in `eclair-kmp` than `eclair`, no need to ensure the transaction is completely valid and we don't do script introspection.